# HG changeset patch
# Parent 95fe59fa4df8614a0bf6b0f5b7c7114e98a28f8d
rhbz#632236: [abrt] [docx] _Construct<long, long> crash

diff -r 95fe59fa4df8 writerfilter/source/dmapper/DomainMapperTableHandler.cxx
--- a/writerfilter/source/dmapper/DomainMapperTableHandler.cxx	Wed Sep 29 09:32:54 2010 +0200
+++ b/writerfilter/source/dmapper/DomainMapperTableHandler.cxx	Wed Sep 29 09:59:15 2010 +0200
@@ -492,13 +492,6 @@
         if( aRepeatIter == m_aTableProperties->end() )
             m_aTableProperties->Insert( PROP_HEADER_ROW_COUNT, false, uno::makeAny( (sal_Int32)0 ));
         
-        // Remove the PROP_HEADER_ROW_COUNT from the table default to avoid
-        // propagating it to the cells
-        PropertyMap::iterator aDefaultRepeatIt =
-        rInfo.pTableDefaults->find( PropertyDefinition( PROP_HEADER_ROW_COUNT, false ) );
-        if ( aDefaultRepeatIt != rInfo.pTableDefaults->end( ) )
-            rInfo.pTableDefaults->erase( aDefaultRepeatIt );
-        
         rInfo.aTableProperties = m_aTableProperties->GetPropertyValues();
         
 #ifdef DEBUG_DOMAINMAPPER
@@ -588,7 +581,14 @@
                     PropertyMapPtr pStyleProps = rInfo.pTableStyle->GetProperties( nCellStyleMask + nRowStyleMask );
                     pAllCellProps->insert( pStyleProps );
                 }
-                
+
+                // Remove properties from style/row that aren't allowed in cells
+                const PropertyMap::iterator aDefaultRepeatIt =
+                    pAllCellProps->find(
+                        PropertyDefinition( PROP_HEADER_ROW_COUNT, false ) );
+                if ( aDefaultRepeatIt != pAllCellProps->end( ) )
+                    pAllCellProps->erase( aDefaultRepeatIt );
+
                 // Then add the cell properties
                 pAllCellProps->insert( *aCellIterator );
                 aCellIterator->get( )->swap( *pAllCellProps.get( ) );
